odl/operator/default_ops.py

```diff
@@ -957,6 +959,20 @@ def _call(self, x):
         """Return ``self(x)``."""
         return x.real
 
+    def derivative(self, x):
+        """Return the derivative operator.
+
+        Note that the returned operator (``self``) is the derivative in the
+        Gâteaux sense, but not in the Fréchet sense since it is not linear.
```

Comment:
Are you sure about the Fréchet and Gâteaux sense? As far as I have learned, even the Gâteaux derivative must by definition be linear in the direction. I would formulate it as follows:

Note that the returned operator (``self``) is the derivative only if the spaces are regraded as real vector spaces. The mapping is not differentiable as a mapping between complex vector spaces.

Reply:
According to [Wikipedia](https://en.wikipedia.org/wiki/G%C3%A2teaux_derivative) linearity is not required, only existence of the limit for all points in question.

Reply:
Yes, you are right, and the Hille–Phillips book works with the same convention. Still, if we regard ℂ^n as a real vector space (which is legitimate), then this is a Fréchet-differentiable mapping and the derivative is calculated correctly.

(By the way, the ``range`` method gives back a real vector space, and I do not know if the derivative of a mapping from a ℂ-vector space to a ℝ-vector space can really be defined.)

Reply:
Thanks for checking a more waterproof source. Regarding the sanity of the approach, I think (to quote the [Zen of Python](https://en.wikipedia.org/wiki/Zen_of_Python)) the "practicality beats purity" principle applies here. For all practical purposes, code that deals with complex vector spaces should not have to explicitly cast to "pair of real". 
So IMO we should make it as correct as possible without interfering with other code for complex vectors.

Reply:
Yeah, I think it gets pretty hairy. After looking again at the Wikipedia text, I come to the conclusion that "proper" Gateaux derivatives in complex spaces requires the limit to be taken for arbitrary paths in the complex plane, which would be equivalent to Cauchy-Riemann, and thus complex differentiability. So in some sense we take the simple approach and look a bit the other way. Everything should make sense when we interpret all complex spaces as larger real spaces.

I'll mention the notions of differentiation in the docs.

Reply:
We have a whole file in the main doc here:

https://odlgroup.github.io/odl/math/derivatives_guide.html

It might be a good idea to update it with stuff like this, mostly for our own sake as a reference